src/couch_replicator/src/couch_replicator_api_wrap.erl

```diff
@@ -59,11 +59,11 @@
 db_uri(#httpdb{url = Url}) ->
     couch_util:url_strip_password(Url);
 
-db_uri(#db{name = Name}) ->
-    db_uri(Name);
+db_uri(DbName) when is_binary(DbName) ->
```

Comment:
@iilyak For clarity on these sorts of clause ordering changes I was trying to keep fairly consistent with the existing logic. Previously this either accepted a `#db{}` or a binary. Though the binary function clause/badmatch would have come from binary_to_list rather the db_uri. So there are subtle changes like that but the domain of acceptable values should be identical.